### PR TITLE
Add ADAL.js usage pattern to readme

### PR DESCRIPTION
**Pull request by navyasric**
@nehaagrawal Please review and merge this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,9 +8,112 @@ This library works with both plain JS as well as AngularJS applications.
...
+
+## Usage
+
+#### Prerequisite
+
+Before using ADAL JS, follow the instructions to [register your application](https://docs.microsoft.com/en-us/azure/active-directory/develop/active-directory-integrating-applications) on the Azure portal. Also, make sure to enable the [OAuth 2.0 implicit flow](https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-oauth2-implicit-grant-flow) by setting the property `oauth2AllowImplicitFlow` to true by editing your application manifest on the portal. Implicit flow is used by ADAL JS to get tokens.
+
+#### 1. Instantiate the AuthenticationContext
+
+Instantiate the global variable AuthenticationContext with a minimal required config of clientID. You can read about other configurable options [here](https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Config-authentication-context#configurable-options-).
```
Comment (nehaagrawal) [inline README.md]: can we remove the - at the end of https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Config-authentication-context#configurable-options ?

File: README.md [hunk 2]
```diff
@@ -8,9 +8,112 @@ This library works with both plain JS as well as AngularJS applications.
...
+window.config = {
+    clientId: '[Enter your client_id here, e.g. g075edef-0efa-453b-997b-de1337c29185]',
+};
+
+var authContext = new AuthenticationContext(config);
+```
+
+#### 2. Login the user
+
+Your app must login the user to establish user context. The login operates in popup mode if you set the option `popUp: true` in the config above.
```
Comment (nehaagrawal) [inline README.md]: can we add the code snippet for popup config above at line 44?

Comment (navyasric) [reply README.md]: Added it

File: README.md (hunk 1 above)
Comment (navyasric) [reply README.md]: Fixed this

File: README.md [hunk 3]
```diff
@@ -8,9 +8,113 @@ This library works with both plain JS as well as AngularJS applications.
...
+    // Initiate login
+    authenticationContext.login();
+}
+```
+
+#### 3. Get an access token
+
+Next, you can get access tokens for the APIs your app needs to call using the acquireToken method which attempts to acquire token silently. The acquireToken method takes a callback function as shown below.
+
+If the silent token acquisition fails for some reasons such as expiration, you will need to invoke one of the interactive methods to acquire tokens.
```
Comment (manoj-rath) [inline README.md]: can you pls clarify something like,
"such as expired session" instead of "such as expiration"

Comment (navyasric) [reply README.md]: Updated

File: README.md [hunk 4]
```diff
@@ -8,9 +8,113 @@ This library works with both plain JS as well as AngularJS applications.
...
+
+Instantiate the global variable AuthenticationContext with a minimal required config of clientID. You can read about other configurable options [here](https://github.com/AzureAD/azure-activedirectory-library-for-js/wiki/Config-authentication-context#configurable-options).
+
+```JavaScript
+window.config = {
+    clientId: '[Enter your client_id here, e.g. g075edef-0efa-453b-997b-de1337c29185]',
+    popUp: true
+};
+
+var authContext = new AuthenticationContext(config);
```
Comment (sliekens) [inline README.md:48]: It should be clarified that you can only do this once. See #827.

**Comment by nehaagrawal** [comment]
We should have waited to have everyone's comment for this PR and get approval. I had more comments.

**Pull request merged**